server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -592,19 +592,33 @@ private ImmutableMap<Path, String> getScalarAnswers(ApplicantQuestion question)
                         selectedOptions.stream().collect(Collectors.joining(", ", "[", "]")))
                 .orElse(""));
       case FILEUPLOAD:
-        return ImmutableMap.of(
-            question.getContextualizedPath().join(Scalar.FILE_KEY),
-            question
-                .createFileUploadQuestion()
-                .getFileKeyValue()
-                .map(
-                    fileKey ->
-                        baseUrl
-                            + controllers.routes.FileController.adminShow(
-                                    programDefinition.id(),
-                                    URLEncoder.encode(fileKey, StandardCharsets.UTF_8))
-                                .url())
-                .orElse(""));
+        FileUploadQuestion fileUploadQuestion = question.createFileUploadQuestion();
+        if (fileUploadQuestion.getFileKeyListValue().isPresent()) {
+          return ImmutableMap.of(
+              question.getContextualizedPath().join(Scalar.FILE_KEY),
+              fileUploadQuestion.getFileKeyListValue().orElse(ImmutableList.of()).stream()
```

Comment:
**question (non-blocking):** Is the `.orElse` reachable since we already checked that a value is present?

Reply:
It is not, but when we rip out the if statement above (when we remove the flag) it will be, and will be very important. This seemed the best approach to me to make that future step easier /safer.